pandas/core/groupby/groupby.py

```diff
@@ -582,6 +582,15 @@ def wrapper(*args, **kwargs):
                kwargs_with_axis['axis'] is None:
                 kwargs_with_axis['axis'] = self.axis
 
+            if (name == 'hist' and
+                    kwargs.pop('equal_bins', False)):
+                # GH-22222
+                bins = kwargs.pop('bins', None)
+                if type(bins) == int:
```

Comment:
Would be preferable I think to use `is_integer` from `pandas.core.dtypes.inference`

Reply:
done